lib/ansible/module_utils/common/dict_transformations.py

```diff
@@ -118,7 +118,10 @@ def dict_merge(a, b):
         if k in result and isinstance(result[k], dict):
             result[k] = dict_merge(result[k], v)
         else:
-            result[k] = deepcopy(v)
+            if isinstance(result[k], list):
+                result[k] = [dict_merge(result[k][i], v[i]) if isinstance(result[k][i], dict) else deepcopy(v[i]) for i in range(len(result[k]))]
+            else:
+                result[k] = deepcopy(v)
```

Comment:
I think this might be a behavior change, and I'm not sure how it might impact code already using this.

cc @jillr @s-hertel

Reply:
azure_rm_appgateway and kubevirt are the only modules other than `k8s` that are using `dict_merge`